client/src/features/SelectFromList.js

```diff
@@ -29,32 +28,29 @@ const MenuProps = {
   },
 };
 
-const SelectFromList = ({ dataList, actionType, setValue, currentValue }) => {
+const SelectFromList = ({ props }) => {
   const classes = useStyles();
-
-  if (dataList === undefined) {
-    dataList = [];
-  }
+  const { listOfCampaigns, actionType, setCampaignId, campaignId } = props;
```

Comment:
I'm switching back from passing-down-props-in-an-object back to destructuring the props in the next push for better code readability: 
({ props }) 
to
({ dataList, actionType, setValue, currentValue })